app/assets/javascripts/shared/datatable/state.js

```diff
@@ -1,53 +1,61 @@
-$(document).on('preInit.dt', 'body.issues.index', function (e, settings) {
-  var $paths = $('[data-behavior~=datatable-paths]');
-  if ($paths.data('table-state-url') == undefined) {
-    return;
-  }
+$(document).on(
+  'preInit.dt',
+  'body.issues.index',
+  function (e, settings) {
+    var $paths = $('[data-behavior~=datatable-paths]');
+    if ($paths.data('table-state-url') == undefined) {
+      return;
+    }
 
-  var api = new $.fn.dataTable.Api(settings);
+    var api = new $.fn.dataTable.Api(settings);
 
-  api.button().add(1, {
-    attr: {
-      'data-behavior': 'table-action',
-    },
-    autoClose: true,
-    className: 'd-none',
-    extend: 'collection',
-    name: 'stateBtn',
-    text: '<i class="fa-solid fa-adjust fa-fw"></i>State<i class="fa-solid fa-caret-down fa-fw"></i>',
-    buttons: DradisDatatable.prototype.setupStateButtons.call(api),
-  });
-});
-
-$(document).on('init.dt', 'body.issues.index', function (e, settings) {
-  var $paths = $('[data-behavior~=datatable-paths]');
-  if ($paths.data('table-state-url') == undefined) {
-    return;
+    api.button().add(1, {
+      attr: {
+        'data-behavior': 'table-action',
+      },
+      autoClose: true,
+      className: 'd-none',
+      extend: 'collection',
+      name: 'stateBtn',
+      text: '<i class="fa-solid fa-adjust fa-fw"></i>State<i class="fa-solid fa-caret-down fa-fw"></i>',
+      buttons: DradisDatatable.prototype.setupStateButtons.call(api),
+    });
   }
+);
+
+$(document).on(
+  'init.dt',
+  'body.issues.index',
+  function (e, settings) {
+    var $paths = $('[data-behavior~=datatable-paths]');
+    if ($paths.data('table-state-url') == undefined) {
+      return;
+    }
 
-  var api = new $.fn.dataTable.Api(settings);
-  api.on(
-    'select.dt deselect.dt',
-    function () {
-      var selectedCount = this.rows({ selected: true }).count();
-      DradisDatatable.prototype.toggleStateBtn.call(api, selectedCount !== 0);
-    }.bind(api)
-  );
-});
+    var api = new $.fn.dataTable.Api(settings);
+    api.on(
+      'select.dt deselect.dt',
+      function () {
+        var selectedCount = this.rows({ selected: true }).count();
+        DradisDatatable.prototype.toggleStateBtn.call(api, selectedCount !== 0);
+      }.bind(api)
+    );
+  }
+);
 
 DradisDatatable.prototype.setupStateButtons = function () {
   var states = $('[data-behavior~=dradis-datatable]').data('state-icons'),
     stateButtons = [],
     api = this;
 
-  if ($('[data-behavior~=qa-viewer]').length > 0) {
+  if ($('[data-behavior~=qa-viewer]').length) {
```

Comment:
same sort of thing. Doesn't belong hard-coded in a shared/ module.

Reply:
Moved the state icon handling to the view ✅